biblib/views/document_view.py

```diff
@@ -540,3 +477,286 @@ def delete(self, library):
             return err(NO_PERMISSION_ERROR)
 
         return {}, 200
+
+class QueryView(BaseView):
+    """
+    End point to interact with a specific library, by adding documents and
+    removing documents.
+    """
+    # TODO: adding tags using PUT for RESTful endpoint?
+
+    decorators = [advertise('scopes', 'rate_limit')]
+    scopes = ['user']
+    rate_limit = [1000, 60*60*24]
+
+    @classmethod
+    def add_query_to_library(cls, library_id, document_data):
+        """
+        Adds a the results of a query to a user's library
+        :param library_id: the library id to update
+        :param document_data: the /search query parameters
+
+        :return: {number_added: number of documents successfully added, valid_bibcodes: corresponding bibcodes}
+        """
+        current_app.logger.info('Adding queried documents: {0} to library_uuid: {1}'
+                                .format(document_data, library_id))
+
+        with current_app.session_scope() as session:
+            # Find the specified library
+            library = session.query(Library).filter_by(id=library_id).one()
+
+            start_length = len(library.bibcode)
+            #Validate supplied bibcodes to confirm they exist in SOLR
+            current_app.logger.info("Calling SOLR Query with params: {}".format(document_data.get('params')))
+            response = cls.standard_ADS_bibcode_query(params=document_data.get('params'))
+            solr_resp = response.json()
+            status_code = response.status_code
+            current_app.logger.info("SOLR response: {}".format(solr_resp))
+            
+            if "error" in solr_resp.keys():
+                #If SOLR request fails, pass the error back to the user
+                current_app.logger.error("Failed to retrieve bibcodes with error: {}".format(solr_resp.get("error")))
+                valid_bibcodes = []
+                output_dict = {"error": solr_resp.get("error"), "number_added": 0, "status": status_code}
+                return output_dict
+            else:
+                current_app.logger.info("docs: {}".format(solr_resp.get('response').get("docs")))
+                #If SOLR query succeeds generate list of valid bibcodes from response
+                valid_bibcodes = [doc.get('bibcode') for doc in solr_resp.get('response').get('docs')]
+                current_app.logger.info("Found the following valid bibcodes: {}".format(valid_bibcodes))
```

Comment:
We're logging the list of bibcodes returned twice - once here and once at line 514 above. For the one at line 514, I'd probably log something shorter, and leave this one, but up to you.

Reply:
Opted to log just solr status code at 514